Sources/Algorithms/LazySplitCollection.swift

```diff
@@ -0,0 +1,280 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// A collection that lazily splits a base collection into subsequences separated by elements that satisfy the given `whereSeparator` predicate.
+///
+/// - Note: This type is the result of `x.split(maxSplits:omittingEmptySubsequences:whereSeparator)` and
+/// `x.split(separator:maxSplits:omittingEmptySubsequences)`, where `x` conforms to `LazyCollection`.
+public struct LazySplitCollection<Base: LazyCollectionProtocol> where Base.Element: Equatable, Base.Elements.Index == Base.Index {
```

Comment:
We don't need a `Base.Element: Equatable` constraint here.

I think it also simplified things a bit if `Base` is simply a `Collection`, and we wrap `self.elements` instead of `self` from `LazyCollectionProtocol.split`, because then we don't need to bother with `Base.Elements` here.

Reply:
Eliminated the unnecessary constraint. I need to ponder your other suggestion some more to make sure I fully understand it.

Reply:
Yes, wrapping a collection instead of a lazy collection did make the code more concise. Thank you.